ocaml/lambda/simplif.ml

```diff
@@ -793,14 +795,20 @@ let split_default_wrapper ~id:fun_id ~kind ~params ~return ~body
         List.iter (fun (id, _) -> if Ident.Set.mem id fv then raise Exit) map;
 
         let inner_id = Ident.create_local (Ident.name fun_id ^ "_inner") in
-        let map_param p layout =
+        let map_param p =
           try
             (* If the param is optional, then it must be a value *)
-            List.assoc p map, Lambda.layout_field
+            (* CR ncourant: is the mode necessarily heap? *)
```

Comment:
I think the mode is currently always heap, so alloc_heap is fine here. However, I think it would be more future-proof to use `p.mode` instead of `alloc_heap` as the mode, since we might support local optional arguments eventually.